### ISSUE TITLE
Sidenav scrolls back to top after open if push or aside

### ISSUE THREAD

**Issue by olingerc** [angular/components#9225]
#### Bug, feature request, or proposal:
If a sidenav has sufficient content to make it scrollable, it does not keep previous scroll position when reopened. In fact the scroll position jumps back to the top but only after the animation has finished. This only happens on modes "over" and "push", not so if "side".

#### What is the expected behavior?
The scroll position stays the same if the sidenav is reopened

#### What is the current behavior?
The scroll position always jumps to the top in modes over and push

#### What are the steps to reproduce?
Using the stackblitz form the api docs [here](https://stackblitz.com/angular/jbpmnlrrqqp?file=app%2Fsidenav-mode-example.html) and filling the sidenav with content until scrolling happens makes the issue visible. Use ESCAPE to close the sidenav after scrolling. Then repoen to view issue.

#### What is the use-case or motivation for changing an existing behavior?
This is a bad user experience.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular Material v 5.0.3

**Comment by crisbeto** [comment]
I think what you're seeing is the sidenav focusing the first focusable element when it's opened (e.g. if you have a button at the top of the sidenav), which causes the browser to scroll it into view.

**Comment by olingerc** [comment]
@crisbeto I think you are correct. However, the whole sidenav contains lots of link tags, so there will always be focusable elements. Do you know how I can turn off the auto focusing? It is not really clear from the documentation.

**Comment by crisbeto** [comment]
It's not possible to turn it off at the moment, because it's an accessibility feature that ensures that focus for keyboard users isn't lost when they open a sidenav. cc @mmalerba

**Comment by olingerc** [comment]
After some thought, I actually agree with this. I read about the cdk-focus-start attribute and this actually gives some customisation but I had to find out about it in the docs.
Another issue is, that the focusing is done AFTER the panel has opened and the user will see the panel jump up. This does not look good at all.

**Comment by crakdelpol** [comment]
I have the same problem, on the documentation i see the attribute "md-disable-scroll-target" that have this description "Selector, pointing to an element, whose scrolling will be disabled when the sidenav is opened. By default this is the sidenav's direct parent."
But it doesn't seem to work.
The temporary solution for me (bad but work) is convert all elements in inputs (stylized look like div) and add property "md-autofocus" on interested input

**Comment by Keips** [comment]
*.ts
@Input() private menuSidenav: MatSidenav;
this.menuSidenav.autoFocus = false;

**Comment by tuanbs** [comment]
Works for me. Thanks @Keips

**Comment by Mds92** [comment]
```
<mat-sidenav #sidenav [mode]="'over'" class="files-sidenav" position="end" [autoFocus]="false">
</mat-sidenav>
```

[autoFocus]="false"

**Comment by ccjmne** [comment]
@crisbeto Hello,
I actually would like to auto-focus some specific element in my sidenav when it opens.

I switched on `[autoFocus]="true"` and dropped the `cdkFocusInitial` directive onto the component I need to be focused. It works just fine... unless the `MatSidenav` is in `mode="side"`.
This seems to be directly linked to the issue discussed here, and you didn't seem to be at all surprised the `side` mode would behave differently.

Is there a reason why autofocus doesn't work in `side` mode, or is it actually an issue I should report?

Thanks!

**Comment by libvirtadept** [comment]
> ```
> <mat-sidenav #sidenav [mode]="'over'" class="files-sidenav" position="end" [autoFocus]="false">
> </mat-sidenav>
> ```
> 
> [autoFocus]="false"

Thank you. It is helped ^_^

**Comment by tommyc38** [comment]
Thank you @Mds92 and @Keips.  I was seriously losing my mind trying to figure out what the heck was going on.

**Comment by jelbourn** [comment]
Closing since this is working as intended, and there is now a setting on the sidenav to control this behavior. I filed #19450 to track the fact that the sidenav docs don't explicitly mention `cdkFocusInitial`.
